pkg/kubelet/kubelet_node_status.go

```diff
@@ -171,6 +175,36 @@ func (kl *Kubelet) reconcileHugePageResource(initialNode, existingNode *v1.Node)
 	return requiresUpdate
 }
 
+// reconcileSwapResource will update swap capacity and remove swap if the feature is disabled
+func (kl *Kubelet) reconcileSwapResource(initialNode, existingNode *v1.Node) bool {
+	capacity, found := existingNode.Status.Capacity[nodev1.ResourceSwap]
+	if found && !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.NodeSwap) {
```

Comment:
does the function currently set swap even if the feature is disabled?

should this be:

```go
if !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.NodeSwap) {
  if !found {
    return false
  }
  // ... remove
  return true
}
```

Reply:
```
	capacity, found := existingNode.Status.Capacity[nodev1.ResourceSwap]
	if !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.NodeSwap) {
		if !found {
			return false
		}
		delete(existingNode.Status.Capacity, nodev1.ResourceSwap)
		delete(existingNode.Status.Allocatable, nodev1.ResourceSwap)
		klog.InfoS("Removing swap resource from node object since feature gate NodeSwap is disabled")
		return true
	}
...
```

Fixed. Thanks.